torch/distributed/algorithms/model_averaging/averagers.py

```diff
@@ -84,10 +88,10 @@ def __init__(
 
     def average_parameters(self):
         r"""
-        Averages parameters if ``step`` is less than ``warmup_steps``,
-        or it can be divided by ``period``, where ``step`` is increased by 1
+        Averages parameters if ``step`` is no less than ``warmup_steps``
+        and it can be divided by ``period``, where ``step`` is increased by 1
         at each iteration in the training loop.
         """
-        if self.step < self.warmup_steps or self.step % self.period == 0:
+        if self.step >= self.warmup_steps and self.step % self.period == 0:
```

Comment:
If warmup steps is 101 and period is 4, is it correct that we don't average for steps 101 - 103 and then average on step 104? Shouldn't this check be `(self.step - self.warmup_steps) % self.period == 0` ?

Reply:
Fixed and updated the test case to cover this case.